src/pages/SingleVideoPage/index.jsx

```diff
@@ -3,30 +3,38 @@ import React, { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
 import axios from "axios";
 import { AiOutlineLike } from "react-icons/ai";
-import {
-  MdOutlineWatchLater,
-  MdOutlineVideoLibrary,
-} from "react-icons/md";
-import { useData } from "../../context";
+import { MdOutlineWatchLater, MdOutlineVideoLibrary } from "react-icons/md";
+import { useData, useAuth } from "../../context";
 import { VideoCard } from "../../components";
 import { getRandomVideos } from "../../utils/";
+import { addToHistory } from "../../utils/services";
 
 const SingleVideoPage = () => {
   const { videoId } = useParams();
   const {
-    dataState: { videos },
+    dataState: { videos, history },
+    dataDispatch,
   } = useData();
 
+  const {
+    authState: { token },
+  } = useAuth();
+
   const [currVideo, setCurrVideo] = useState({});
 
   useEffect(() => {
     (async () => {
-      const response = await axios.get(`/api/video/${videoId}`);
-      if (response.status === 200) setCurrVideo(response.data.video);
+      const {data, status} = await axios.get(`/api/video/${videoId}`);
+      if (status === 200) setCurrVideo(data.video);
+      const isInHistory = history.find((curr) => curr._id === videoId);
+      if (token && !isInHistory) addToHistory(data.video, token, dataDispatch);
     })();
+
     return () => setCurrVideo({});
   }, [videoId]);
 
+  useEffect(() => {});
```

Comment:
Can remove the empty useEffect If not used

Reply:
thanks mani, removed it